### PR TITLE
Metrics for bindings

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
Provides the following metrics:
- kcp_keb_v2_bind_duration_millisecond
- kcp_keb_v2_unbind_duration_millisecond
- kcp_keb_v2_binding_created_total
- kcp_keb_v2_minutes_since_earliest_binding_expiration

**Related issue(s)**
Related: https://github.com/kyma-project/kyma-environment-broker/issues/1359

### CODE REVIEW

File: internal/metricsv2/bind.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+		if recovery := recover(); recovery != nil {
+			c.logger.Errorf("panic recovered while handling in progress operation counter: %v", recovery)
+		}
+	}()
+
+	if err := c.updateMetrics(); err != nil {
+		c.logger.Error("failed to update metrics metrics", err)
+	}
+
+	ticker := time.NewTicker(c.poolingInterval)
```
Comment (jaroslaw-pieszka) [inline internal/metricsv2/bind.go:138]: Is it about polling or pooling?

File: internal/storage/postsql/read.go [hunk 2]
```diff
@@ -1053,6 +1053,20 @@ func (r readSession) ListInstancesArchived(filter dbmodel.InstanceFilter) ([]dbm
 	return instancesArchived, len(instancesArchived), totalCount, nil
 }
 
+func (r readSession) GetBindingsStatistics() (dbmodel.BindingStats, error) {
+	dto := dbmodel.BindingStats{}
+	statement := r.session.Select("max(extract(epoch from AGE(now(), expires_at))) as max_expiration_time_in_seconds").From(BindingsTableName)
```
Comment (jaroslaw-pieszka) [inline internal/storage/postsql/read.go:1058]: This is maximum expiration in seconds observed in this point of time. Am I correct?
Not greater than currently set maximum expiration timespan.

File: internal/storage/dbmodel/binding.go [hunk 3]
```diff
@@ -15,3 +15,7 @@ type BindingDTO struct {
 	ExpirationSeconds int64
 	CreatedBy         string
 }
+
+type BindingStats struct {
```
Comment (jaroslaw-pieszka) [inline internal/storage/dbmodel/binding.go:19]: Why do we use the same name `BindingStats` for two different types?

File: internal/metricsv2/bind.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+func NewBindingStatsCollector(db storage.Bindings, poolingInterval time.Duration, logger logrus.FieldLogger) *BindingStatitics {
+	return &BindingStatitics{
+		db:              db,
+		logger:          logger,
+		poolingInterval: poolingInterval,
+		MinutesSinceEarliestExpirationMetric: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: prometheusNamespacev2,
+			Subsystem: prometheusSubsystemv2,
+			Name:      "minutes_since_earliest_binding_expiration",
+			Help:      "Specifies the time in minutes since the earliest binding expiration.",
```
Comment (piotrmiskiewicz) [inline internal/metricsv2/bind.go:117]: add more explanation why the metrics is created, how to read it

File: internal/storage/dbmodel/binding.go [hunk 5]
```diff
@@ -15,3 +15,7 @@ type BindingDTO struct {
...
+	MaxExpirationTimeInSeconds *float64 `db:"max_expiration_time_in_seconds"`
```
Comment (piotrmiskiewicz) [inline internal/storage/dbmodel/binding.go:20]: use the correct name

File: internal/storage/dbmodel/binding.go (hunk 3 above)
Comment (piotrmiskiewicz) [reply internal/storage/dbmodel/binding.go:19]: let's use 1 type

File: cmd/broker/suite_test.go [hunk 6]
```diff
@@ -941,27 +943,32 @@ func fixConfig() *Config {
...
+		Provisioning:                              process.StagedManagerConfiguration{MaxStepProcessingTime: time.Minute},
+		Deprovisioning:                            process.StagedManagerConfiguration{MaxStepProcessingTime: time.Minute},
+		Update:                                    process.StagedManagerConfiguration{MaxStepProcessingTime: time.Minute},
+		ArchiveEnabled:                            true,
+		CleaningEnabled:                           true,
+		UpdateRuntimeResourceDelay:                time.Millisecond,
+		MetricsV2: metricsv2.Config{
+			Enabled:                                         true,
+			OperationResultRetentionPeriod:                  time.Hour,
+			OperationResultPoolingInterval:                  3 * time.Second,
```
Comment (jaroslaw-pieszka) [inline cmd/broker/suite_test.go:967]: Should be `polling`

File: internal/metricsv2/operations_result.go [hunk 7]
```diff
@@ -41,7 +41,7 @@ func NewOperationResult(ctx context.Context, db storage.Operations, cfg Config,
 			Name:      "operation_result",
 			Help:      "Results of metrics",
 		}, []string{"operation_id", "instance_id", "global_account_id", "plan_id", "type", "state", "error_category", "error_reason", "error"}),
-		poolingInterval:                  cfg.OperationResultPoolingInterval,
+		poolingInterval:                  cfg.OperationResultPollingInterval,
```
Comment (jaroslaw-pieszka) [inline internal/metricsv2/operations_result.go:44]:
```suggestion
		pollingInterval:                  cfg.OperationResultPollingInterval,
```

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**